doc/source/hclust.rst

```diff
@@ -0,0 +1,41 @@
+Hierarchical Clustering
+========================
+
+`Hierarchical clustering <https://en.wikipedia.org/wiki/Hierarchical_clustering>`_ algorithms build a dendrogram of nested clusters by repeatedly merging or splitting clusters.
+
+**Functions**
+
+.. function:: hclust(D, method)
+
+	Perform hierarchical clustering on distance matrix D with specified method.
+
+	:param D: The pairwise distance matrix. ``D[i,j]`` is the distance between points ``i`` and ``j``.
+	:param method: A Symbol specifying how distance is measured between clusters (which is used to determine which clusters to merge on each iteration). Valid methods are ``:single``, ``:average``, or ``:complete``.
```

Comment:
Valid methods are listed below. To avoid duplication I suggest to remove the last sentence.